xfields/ibs/_equilibrium.py

```diff
@@ -253,191 +253,207 @@ def compute_emittance_evolution(
+    # if gemitt_x is not None:
+    #     assert nemitt_x is None, "Cannot provide both 'gemitt_x' and 'nemitt_x'"
+    # if gemitt_y is not None:
+    #     assert nemitt_y is None, "Cannot provide both 'gemitt_y' and 'nemitt_y'"
+    # if gemitt_zeta is not None:
+    #     assert nemitt_zeta is None, "Cannot provide both 'gemitt_zeta' and 'nemitt_zeta'"
+    # # ---------------------------------------------------------------------------------------------
+    # # Convert emittances to geometric if needed
+    # if nemitt_x is not None:
+    #     assert gemitt_x is None, "Cannot provide both 'gemitt_x' and 'nemitt_x'"
+    #     gemitt_x = nemitt_x / (twiss.beta0 * twiss.gamma0)
+    # if nemitt_y is not None:
+    #     assert gemitt_y is None, "Cannot provide both 'gemitt_y' and 'nemitt_y'"
+    #     gemitt_y = nemitt_y / (twiss.beta0 * twiss.gamma0)
+    # if nemitt_zeta is not None:
+    #     assert gemitt_zeta is None, "Cannot provide both 'gemitt_zeta' and 'nemitt_zeta'"
+    #     gemitt_zeta = nemitt_zeta / (twiss.beta0 * twiss.gamma0)
+    # ---------------------------------------------------------------------------------------------
+    # Check for valid value of emittance_constraint and warn if constraint provided but factor is 0
+    if emittance_constraint is not None:
+        assert emittance_constraint.lower() in ("coupling", "excitation"), "Invalid 'emittance_constraint', accepted values are 'coupling' or 'excitation'."
+        if emittance_coupling_factor == 0:
+            LOGGER.warning("As 'emittance_coupling_factor` is zero, providing 'emittance_constraint' has no effect!")
+    # ---------------------------------------------------------------------------------------------
     # Handle initial transverse emittances and potential effect of coupling / excitation constraints
     # TODO: I don't like this, would rather force the user to provide gemitt_x, gemitt_y & gemitt_zeta
     if initial_emittances is None:
-        print("Emittances from the Twiss object are being used.")
+        _print("Emittances from the Twiss object are being used.")
         emittance_x, emittance_y, emittance_z = (
             twiss.eq_gemitt_x,
             twiss.eq_gemitt_y,
             twiss.eq_gemitt_zeta,
         )
-        # If emittance_coupling_factor is non zero, then natural emittance is
-        # modified accordingly
-        # fmt: off
-        if emittance_coupling_factor != 0 and emittance_constraint.lower() == "coupling":
-            # The convention used is valid for arbitrary damping partition numbers and emittance_coupling_factor.
+        # If emittance_coupling_factor is non zero, then natural emittance is modified accordingly (used
+        # convention is valid for arbitrary damping partition numbers and emittance_coupling_factor).
+        if emittance_constraint.lower() == "coupling" and emittance_coupling_factor != 0:
```

Comment:
Cosmetic. Same two checks but I find checking for the emittance_constraint first makes it easier to read and understand the code flow.